File: algo-binary-search/python/binary_search.py
def binary_search(num, array, start, end):
    sorted_arr = sorted(array)
    if end >= start:
        middle = int(start + end) // 2
        if sorted_arr[middle] == num:
            return middle
        elif num < sorted_arr[middle]:
            return binary_search(num, sorted_arr, start, middle - 1)
        else:
            return binary_search(num, sorted_arr, middle + 1, end)
    else:
        return -1

File: algo-binary-search/js/binarySearch.js
function binarySearchRecursive(value, arr, start, stop){

  let middle = start + Math.floor((stop-start)/ 2);
  if (stop < start){
    return -1;
  }

  if (arr[middle] == value) {
    return middle;
  } else if (value < middle) {
    return binarySearchRecursive(value, arr, start, middle -1);
  } else if (value > middle) {
    return binarySearchRecursive(value, arr, middle + 1, stop);
  }
}

function binarySearch(value, arr){
  return binarySearchRecursive(value, arr.sort(), 0, arr.length-1);
}